libs/libarchfpga/src/read_fpga_interchange_arch.cpp

```diff
@@ -811,14 +885,11 @@ struct ArchReader {
     void process_constants() {
         auto consts = ar_.getConstants();
 
-        arch_->gnd_cell = str(consts.getGndCellType());
-        arch_->vcc_cell = str(consts.getVccCellType());
+        arch_->gnd_cell = std::make_pair(str(consts.getGndCellType()), str(consts.getGndCellPin()));
+        arch_->vcc_cell = std::make_pair(str(consts.getVccCellType()), str(consts.getVccCellPin()));
 
-        if (consts.getGndNetName().isName())
-            arch_->gnd_net = str(consts.getGndNetName().getName());
-
-        if (consts.getVccNetName().isName())
-            arch_->vcc_net = str(consts.getVccNetName().getName());
+        arch_->gnd_net = consts.getGndNetName().isName() ? str(consts.getGndNetName().getName()) : "$__gnd_net";
```

Comment:
Are these hard-coded const net names special in any way? Maybe we could define them as constants somewhere else in the code?

Reply:
So, in general the const names should be present in the interchange device. This is just a way to allow having them blank. At the moment there is no interchange device validator and there is no rule to determine whether the const `GND` and `VCC` net need to be defined.

We could opt for erroring out if those are not present